core/common/src/rpc/core/RpcRequest.ts

```diff
@@ -484,6 +499,16 @@ export abstract class RpcRequest<TResponse = any> {
     RpcRequest.events.raiseEvent(RpcRequestEvent.PendingUpdateReceived, this);
   }
 
+  private handleTransientError(status: RpcRequestStatus) {
+    if (!this._active)
+      return;
+
+    this.setLastUpdatedTime();
+    this._retryAfter = this.computeRetryAfter(this._attempts - 1);
```

Comment:
@swbsi do we really want to retry these requests indefinitely?  Seems like we should set a max number of attempts, and fail the request after that...

Reply:
Not sure I'm in favor of failing the request in that scenario. Maybe a status listener (aggregating all requests in a router-initiated pending state) that the app can use -- perhaps to notify the user that things are going slowly?

Reply:
Really?  If I `await` some request, it seems obvious to me that it should _eventually_ either succeed or fail, not just spin _forever_...

We can keep the max attempts fairly high if that's a concern, but why shouldn't there be **some** upper bound?

Reply:
But these are not errors. Why isn't a higher level handler appropriate?

Reply:
Of course these are error.  We should retry them, because presume they are like to be transient and thus will eventually resolve themselves, but they're still errors.  If they persist for more than a few minutes, they likely indicate a more serious outage,.

Microsoft clearly identifies endless retry mechanisms as an anti-pattern in their [transient fault handling guide](https://docs.microsoft.com/en-us/azure/architecture/best-practices/transient-faults).

Reply:
The application will definitely want to know if scale-out is going well...but I don't think each individual call site is where it should happen. It seems cumbersome.

Reply:
That msft doc is about actual transient errors...but we are using these codes for scale out status. Perhaps that is where the tension is here...

Reply:
I agree with you that we should fail the request if we are getting essentially a "not available" message.

Reply:
But, the common case is going to be that we are just receiving updates on the progress of scale out. I don't think we should ever fail a request because of that.

Reply:
The orchestrator team chose these responses because this is expected to be a rare, _exceptional_ response.  Crucially, there are potential error conditions where orchestrator components could endlessly send one of these responses without ever actually scaling.  Basically, we would not have chosen an HTTP 5xx status code if the response meant everything was completely and totally hunky-dory.

To say these are not really transient errors because they're expected would be like saying requests that get rejected/rate limited due to throttling - after all, in that case the throttling is working as expected!

IMO, scaling out should take on the order of a few minutes, ergo we should stop retrying sometime around then.

Reply:
Adding @m1o1 for the orchestrator team's perspective

Reply:
It was my understanding that these transient statuses would be very common...since we will be sending them during the normal startup of a backend instance (getting the briefcase, etc) until it is ready for the frontend to use.

Sounds like we need to clarify that :-)

Reply:
My first thought is that if the Retry-After header is missing, then there should be an upper bound on the number of retries. Otherwise, it can keep chugging along. I'll ask the rest of the team what they think too.

Reply:
> My first thought is that if the Retry-After header is missing, then there should be an upper bound on the number of retries. Otherwise, it can keep chugging along. I'll ask the rest of the team what they think too.

I like that approach.

Reply:
Yeah, that sounds good to me too